blackmarket/UI/ViewModels/SignUpViewModel.swift

```diff
@@ -83,9 +86,17 @@ class SignUpViewModel: ObservableObject, Identifiable {
         self.authServices = authServices
     }
     
-    func signUp() async {
-        //TODO: handle errors
-        await authServices.signup(email: emailConfiguration.value,name: nameConfiguration.value, password: passwordConfiguration.value)
+    @MainActor func signUp() async {
```

Comment:
Maybe we can use `.receive(on: DispatchQueue.main)` just with the variables that modify the UI?, because this `@MainActor` is including the whole function with the service call too 
you can do something like: 
```
private let isFetchingSubject = CurrentValueSubject<Bool, Never>(false)
var isFetchingPublisher: AnyPublisher<Bool, Never> {
    isFetchingSubject
      .receive(on: DispatchQueue.main)
      .eraseToAnyPublisher()
  }
```

Reply:
Same in the login function above 👍🏻